stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -288,6 +288,18 @@ function(_add_target_variant_c_compile_flags)
     endif()
   endif()
 
+  # Avoid the need for the FP16 truncation and extension routines from
+  # compiler-rt by assuming that we have hardware capable of performing
+  # half-precision floating point conversions.  This was announced circa 2007,
+  # and is standard with ~Haswell (~2013).
+  if(CFLAGS_ARCH STREQUAL x86_64)
+    if(SWIFT_COMPILER_IS_MSVC_LIKE)
+      list(APPEND result /clang:-mf16c)
+    else()
+      list(APPEND result -mf16c)
+    endif()
+  endif()
```

Comment:
Why not simply combine this with the x86_64 block just above?

Reply:
We absolutely could, it was more a matter of convenience and isolation.  Having it be interleaved is slightly annoying to update in the future.  I'm hoping that as we try to clean up the CMake rules, we can get this all cleaned up.